modules/juce_audio_devices/native/juce_win32_Midi.cpp

```diff
@@ -798,54 +887,69 @@ class WinRTMidiService  : public MidiServiceType
             if (! isEnabled)
                 return S_OK;
 
-            const ScopedLock lock (deviceChanges);
+            {
+                const ScopedLock lock (deviceChanges);
 
-            DeviceInfo info;
+                DeviceInfo info;
 
-            HSTRING name;
-            hr = addedDeviceInfo->get_Name (&name);
+                HSTRING name;
+                hr = addedDeviceInfo->get_Name (&name);
 
-            if (FAILED (hr))
-                return S_OK;
+                if (FAILED (hr))
+                    return S_OK;
 
-            info.name = WinRTWrapper::getInstance()->hStringToString (name);
+                info.name = WinRTWrapper::getInstance()->hStringToString (name);
 
-            HSTRING id;
-            hr = addedDeviceInfo->get_Id (&id);
+                HSTRING id;
+                hr = addedDeviceInfo->get_Id (&id);
 
-            if (FAILED (hr))
-                return S_OK;
+                if (FAILED (hr))
+                    return S_OK;
 
-            info.id = WinRTWrapper::getInstance()->hStringToString (id);
+                info.id = WinRTWrapper::getInstance()->hStringToString (id);
 
-            boolean isDefault;
-            hr = addedDeviceInfo->get_IsDefault (&isDefault);
+                boolean isDefault;
+                hr = addedDeviceInfo->get_IsDefault (&isDefault);
 
-            if (FAILED (hr))
-                return S_OK;
+                if (FAILED (hr))
+                    return S_OK;
+
+                info.isDefault = isDefault != 0;
+                connectedDevices.add (info);
+            }

```

Comment:
The block above is just the original code put in a scope block with extra indentation to make sure the lock is removed before notifying listeners (see lines below), so it does not need to be reviewed.